src/orquestra/sdk/_base/_dsl.py

```diff
@@ -159,6 +159,14 @@ def GithubImport(
                 f"not {type(personal_access_token).__name__}."
             )
         raise TypeError(errmsg)
+    if personal_access_token is not None and isinstance(personal_access_token, Secret):
+        if Secret.workspace_id is None:
```

Comment:
**issue**: `Secret` is a class. What's `Secret.workspace_id`? Also, the unit tests are missing for this branch.